### ISSUE TITLE
StrawberryShake 13.0.5 is not including GraphQLException Error Details in Reponse

### ISSUE THREAD

**Issue by KennethHHughes** [ChilliCream/graphql-platform#5897]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Product

Strawberry Shake

### Describe the bug

This appears to be related to #5802, but is still happening in 13.0.5.

Background: We were running Hot Chocolate and Strawberry Shake version 12.16.0. We have an IHttpRequestInterceptor for authentication that throws a GraphQLException when our JWT is invalid. We have covered this with several integration tests.

After following your migration documentation to get everything updated to 13.0.5 in both Hot Chocolate and Strawberry Shake, our integration tests around this are now failing because the error is not hydrated in the same was as it was in 12.16.0. You can see in the below screenshots that before the update we got back our specific Error details from the GraphQLException we threw in our interceptor. The Code and Message are populated. After the update, the exact same test only has a generic 500 error message with no code populated.

When we hit the same endpoint with a bad JWT from Banana Cake Pop, we still get all the errors details we expected.

Before (12.16.0) - with Error Code and Message
![StrawberryShake_v12 16 0_Response](https://user-images.githubusercontent.com/19213890/221949176-d733d649-ed45-4eeb-b71b-fe5d6bed5421.PNG)

After (13.0.5) - generic message without code
![StrawberryShake_v13 0 5_Response](https://user-images.githubusercontent.com/19213890/221949192-bfc1f42a-11a6-4b6e-85bc-81b1ce286a06.PNG)

### Steps to reproduce

1. Write an IHttpRequestInterceptor that throws a GraphQLException something like this:
```
public class TestInterceptor : DefaultHttpRequestInterceptor
{
    public override async ValueTask OnCreateAsync(
        HttpContext context,
        IRequestExecutor requestExecutor,
        IQueryRequestBuilder requestBuilder,
        CancellationToken cancellationToken)
    {
        await base.OnCreateAsync(context, requestExecutor, requestBuilder, cancellationToken);
        throw new GraphQLException(new Error(
            "Authentication Failed",
            code: "401", 
            exception: new Exception("Failed to authenticate")
        ));
    }
}
```

2. Use `.AddHttpRequestInterceptor(...)` to register the interceptor.
3. Make an HTTP request using StrawberryShake and inspect the result. Notice the details from the Error provided to the GraphQL exception are not present.

### Version

13.0.5

**Comment by michaelstaib** [comment]
Yes, we need to implement the new transport spec for strawberry shake.

**Comment by KennethHHughes** [comment]
Thanks for looking at this @michaelstaib. I do one question to help my understanding. The website indicates that version 13 is now the stable build, but if I'm understanding you correctly, it sounds like version 13 isn't quite to feature parity with version 12. Did I misunderstand that version 13 is the current stable build?

**Comment by colewerner** [comment]
I'm having this same issue with graphql requests that are failing.  I get a generic 500 internal server error.

**Comment by johntyrrell** [comment]
Errors in our AggregateError returned by the IErrorFilter are also not included. 
We also get the generic 500 error

**Comment by xyan0001** [comment]
Hi @michaelstaib may I know if you have any timeline on this issue? Is StrawberryShake v13.0.5 ready for production? Thanks.

**Comment by michaelstaib** [comment]
Yes, around 4 weeks until 13.1

**Comment by xyan0001** [comment]
Hi @michaelstaib  any updates on this? Thanks.
